### DISCUSSION TITLE
How to include changelog for a Helm chart without a GitHub release

### DISCUSSION THREAD

**Discussion by lindhe** [renovatebot/renovate#30029] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Renovate v37.421.2 running on Kubernetes working towards GitHub Enterprise

### Please tell us more about your question or problem

For most of the charts I use, Renovate is able to correctly identify the changelog based on the GitHub release matching the Helm chart's version. Sometimes, I just need to nudge it in the right direction by providing the `sourceUrl`, for whatever reason.

But with some charts, for example [netapp/trident](https://github.com/NetApp/trident) and [bitnami-labs/sealed-secrets](https://github.com/bitnami-labs/sealed-secrets), things are harder! Renovate cannot detect the changelogs for these charts, because there is no matching GitHub release.

* In the case of Trident, there is only a release made for the Trident software and the Helm chart is silently updated and pushed to the Helm registry whenever there is a new version of the software.
* In the case of SealedSecrets, there is a GitHub release but it contains almost no info! To find the correct changelog, one has to first checkout the release of the Helm chart (e.g. [v2.16.0](https://github.com/bitnami-labs/sealed-secrets/releases/tag/helm-v2.16.0)) and then pattern match [the `appVersion` field of the `Chart.yaml`](https://github.com/bitnami-labs/sealed-secrets/blob/e10b787614e5a0c9f523d955d4b3dbf2b09d81b6/helm/sealed-secrets/Chart.yaml#L4) with the corresponding GitHub release (in my example: [v0.27.0](https://github.com/bitnami-labs/sealed-secrets/releases/tag/v0.27.0)).

Clearly, these (and all other odd release processes) are hard for Renovate to handle properly. And I think it's probably too much work and high complexity to try and implement every edge case in Renovate. Sometimes, I'm able to conjure up some fix in `renovate.json`, but sometimes these cases are just too complex to do that.

I'm a frequent contributor to many projects. I would like to help them make charts that are easier for Renovate to parse. But I don't know what to suggest. **Is there anything I can suggest to the project maintainers that would help Renovate find the relevant changelog for these Helm charts?**

I'm hesitant to suggest changing their release processes, that is often a complex thing to change and can be hard to do for various internal reasons. I'm hoping there is something easier I can suggest to them, like for example if Renovate can check an annotation in the `Chart.yaml` with a link to the changelog, or perhaps if there is a `CHANGELOG.md` included in the chart.

Do you have any suggestions or guidelines for how to properly include changelogs?

### Logs (if relevant)

n/a

**Comment by HonkingGoose** [comment, marked as answer]
The [Renovate Docs, key concepts, changelogs](https://docs.renovatebot.com/key-concepts/changelogs/#advice-for-package-maintainers) explains how maintainers can help Renovate "see" their changelogs.

**Reply by lindhe** [reply]
I see. Thanks for the answer.

Since it's not mentioned in the docs, I assume that a CHANGELOG.md in a Helm chart, [like Bitnami usually attach](https://github.com/bitnami/charts/blob/8eb529978d6b43100dd43a7dc2d4a2bfaf0f4ed7/bitnami/apache/CHANGELOG.md), does not get parsed then?

**Reply by lindhe** [reply]
I assume it is not possible to configure Renovate to consider `appVersion` rather than `version` when detecting changelogs?

**Reply by usmonster** [reply]
> Since it's not mentioned in the docs, I assume that a CHANGELOG.md in a Helm chart, [like Bitnami usually attach](https://github.com/bitnami/charts/blob/8eb529978d6b43100dd43a7dc2d4a2bfaf0f4ed7/bitnami/apache/CHANGELOG.md), does not get parsed then?

I believe it should, cf. [this example](https://github.com/usmonster/renovate-test-helm-charts/pull/16).
